addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -290,66 +290,80 @@ def _l10n_it_is_simplified_document_type(self, document_type):
     # -------------------------------------------------------------------------
 
     def _cron_receive_fattura_pa(self):
-        ''' Check the proxy for incoming invoices.
+        ''' Check the proxy for incoming invoices for all companies.
         '''
-        proxy_users = self.env['account_edi_proxy_client.user'].search([('edi_format_id', '=', self.env.ref('l10n_it_edi.edi_fatturaPA').id)])
-
-        if proxy_users._get_demo_state() == 'demo':
+        if self.env['account_edi_proxy_client.user']._get_demo_state() == 'demo':
             return
 
-        for proxy_user in proxy_users:
-            company = proxy_user.company_id
-            try:
-                res = proxy_user._make_request(proxy_user._get_server_url() + '/api/l10n_it_edi/1/in/RicezioneInvoice',
-                                               params={'recipient_codice_fiscale': company.l10n_it_codice_fiscale})
-            except AccountEdiProxyError as e:
-                res = {}
-                _logger.error('Error while receiving file from SdiCoop: %s', e)
+        for proxy_user in self.env['account_edi_proxy_client.user'].search([('edi_format_code', '=', 'fattura_pa')]):
+            self._receive_fattura_pa(proxy_user)
 
-            proxy_acks = []
-            for id_transaction, fattura in res.items():
-                if self.env['ir.attachment'].search([('name', '=', fattura['filename']), ('res_model', '=', 'account.move')], limit=1):
-                    # name should be unique, the invoice already exists
-                    _logger.info('E-invoice already exists: %s', fattura['filename'])
-                    proxy_acks.append(id_transaction)
-                    continue
+    def _save_incoming_attachment_fattura_pa(self, proxy_user, id_transaction, filename, key):
```

Comment:
maybe in the order of methods _save_incoming_attachment... can be put under _receive_fatturapa (as the receive is called first by the cron)